cortex-r-semihosting/src/hio.rs

```diff
@@ -0,0 +1,69 @@
+//! Host I/O
+
+// Fixing this lint requires a breaking change that does not add much value
+#![allow(clippy::result_unit_err)]
```

Comment:
This is a newly released crate, can we make that breaking change so we can drop this?